lib/src/data/communication/api/api_exception_mapper.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:io';
+
+import 'package:allthenews/src/domain/communication/api_exception.dart';
+import 'package:allthenews/src/domain/communication/exception_mapper.dart';
+import 'package:dio/dio.dart';
+
+class ApiExceptionMapper extends ExceptionMapper {
+  @override
+  ApiException toExceptionType(Object error) {
```

Comment:
chyba powinno być Exception zamiast Object, jak wcześniej

Reply:
FutureBuilder zwraca error jako Object i dlatego potem jest sprawdzanie czy to DioError czy coś innego. Jak będzie to wyniesione do Blocu/ChangeNotifiera i nie będzie trzeba polegać na futurebuilderze to ten typ się pewnie też zmieni

Reply:
hmm... ale to wtedy będzie ExceptionMapper będzie miał sygnaturę `Exception toExceptionType(Exception error)`? :P

Reply:
hmm myślę, że ten mapper w ogóle będzie wyjebany w przypadku wyniesienia tego poziom wyżej, zrobiłbym wtedy tam try catch i od razu przy określonym exceptionie zwracał domenowy typ